R/utils.R

```diff
@@ -380,3 +380,39 @@ defunction <- function(x) {
 get_unique_labels <- function(labels) {
   make.unique(gsub("[^[:alnum:]]", "_", tolower(labels)), sep = "_")
 }
+
+
+#' Create `pak` `.lock` file
+#'
+#' Function detects used packages in the session, extends options('repos') with custom repositories,
+#' and then executes `pak::lockfile_create()` that creates a `.lock` file for future session reproducibility.
+#'
+#' @param lockfile lock file name passed to `pak::lockfile_create()`
+#'
+#' @return (`character(n)`) content of `lockfile`.
+#'
+#' @keywords internal
+create_lockfile <- function(lockfile = 'pkg.lock') {
+  # TODO: find a replacement for devtools
+  package_source <- devtools::session_info()$packages$source
```

Comment:
I need an alternative for `devtools` so it does not end up in our Imports dependencies.
The need in here is to detect all addresses of custom R repositories from which packages were installed.

`pak::lockfile_create` operates on `options("repos")` and those custom repositories need to be included in there before you run `pak::lockfile_create`.

In our case custom repositories are e.g. `https://pharmaverse.r-universe.dev`

Reply:
We can use `sessioninfo::session_info` instead of `devtools::session_info` which is a wrapper around `sessioninfo::session_info`